addons/mail/static/src/core/web_portal/message_patch.js

```diff
@@ -154,5 +166,23 @@ patch(Message.prototype, {
             });
             updateFromState();
         }
+        return () => {
+            for (const button of buttons) {
+                for (const [k, v] of Object.entries(buttonStyles)) {
+                    if (!["width", "height"].includes(k)) {
+                        button.style.setProperty(k, v);
+                    }
+                }
+                button.style.visibility = "visible";
+            }
+            for (const icon of icons) {
+                for (const [k, v] of Object.entries(iconStyles)) {
+                    if (!["width", "height"].includes(k)) {
+                        icon.style.setProperty(k, v);
+                    }
+                }
+                icon.style.visibility = "visible";
+            }
+        };
```

Comment:
`prepareMessageBody()` becomes quite complex and very opinionated to the need with shadow DOM, with the return value being reserved for this use-case.

I would suggest to instead register the buttons and icons with styles in dedicated attributes, and call an explicitly named function to restore/enforce the style in the shadow DOM.